pkg/cli/rsync/fsnotification/fsnotification.go

```diff
@@ -26,6 +26,9 @@ func AddRecursiveWatch(watcher *fsnotify.Watcher, path string) error {
 	}
 
 	folders, err := getSubFolders(path)
+	if err != nil {
```

Comment:
This changes the functionality in case only a partial list of paths gets returned. Not sure if that was the intention or whether the check was forgotten.

Reply:
Functionality introduced in https://github.com/openshift/origin/pull/8268

Reply:
AFAICS, @ingvagabund is right, this function is designed to work in cases when partial list is returned. Instead changing the functionality by immediate returning error, I'd suggest add a warning message if there is any error with high verbosity level?;

```
klog.V(5).Infof("error occured while getting sub folders %v", err)
```